Mathlib/GroupTheory/Perm/Support.lean

```diff
@@ -0,0 +1,670 @@
+def support (f : Perm α) : Finset α :=
+  univ.filter fun x => f x ≠ x
+#align equiv.perm.support Equiv.Perm.support
+
+@[simp]
+theorem mem_support {x : α} : x ∈ f.support ↔ f x ≠ x := by
+  rw [support, mem_filter, and_iff_right (mem_univ x), decide_eq_true_iff]
+#align equiv.perm.mem_support Equiv.Perm.mem_support
+
+theorem not_mem_support {x : α} : x ∉ f.support ↔ f x = x := by simp
+#align equiv.perm.not_mem_support Equiv.Perm.not_mem_support
+
+theorem coe_support_eq_set_support (f : Perm α) : (f.support : Set α) = { x | f x ≠ x } := by
+  ext
+  simp
+#align equiv.perm.coe_support_eq_set_support Equiv.Perm.coe_support_eq_set_support
+
+@[simp]
+theorem support_eq_empty_iff {σ : Perm α} : σ.support = ∅ ↔ σ = 1 := by
+  simp_rw [Finset.ext_iff, mem_support, Finset.not_mem_empty, iff_false_iff, not_not,
+    Equiv.Perm.ext_iff, one_apply]
+#align equiv.perm.support_eq_empty_iff Equiv.Perm.support_eq_empty_iff
+
+@[simp]
+theorem support_one : (1 : Perm α).support = ∅ := by rw [support_eq_empty_iff]
+#align equiv.perm.support_one Equiv.Perm.support_one
+
+@[simp]
+theorem support_refl : support (Equiv.refl α) = ∅ :=
+  support_one
+#align equiv.perm.support_refl Equiv.Perm.support_refl
+
+theorem support_congr (h : f.support ⊆ g.support) (h' : ∀ x ∈ g.support, f x = g x) : f = g := by
+  ext x
+  by_cases hx : x ∈ g.support
+  · exact h' x hx
+  · rw [not_mem_support.mp hx, ← not_mem_support]
+    exact fun H => hx (h H)
+#align equiv.perm.support_congr Equiv.Perm.support_congr
+
+theorem support_mul_le (f g : Perm α) : (f * g).support ≤ f.support ⊔ g.support := fun x => by
+  simp only [sup_eq_union]
+  rw [mem_union, mem_support, mem_support, mem_support, mul_apply, ← not_and_or, not_imp_not]
+  rintro ⟨hf, hg⟩
+  rw [hg, hf]
+#align equiv.perm.support_mul_le Equiv.Perm.support_mul_le
+
+theorem exists_mem_support_of_mem_support_prod {l : List (Perm α)} {x : α}
+    (hx : x ∈ l.prod.support) : ∃ f : Perm α, f ∈ l ∧ x ∈ f.support := by
+  contrapose! hx
+  simp_rw [mem_support, not_not] at hx⊢
+  induction' l with f l ih
+  · rfl
+  · rw [List.prod_cons, mul_apply, ih, hx]
+    simp only [List.find?, List.mem_cons, true_or]
+    intros f' hf'
+    refine' hx f' _
+    simp only [List.find?, List.mem_cons]
+    exact Or.inr hf'
+#align
+  equiv.perm.exists_mem_support_of_mem_support_prod
+  Equiv.Perm.exists_mem_support_of_mem_support_prod
+
+theorem support_pow_le (σ : Perm α) (n : ℕ) : (σ ^ n).support ≤ σ.support := fun _ h1 =>
+  mem_support.mpr fun h2 => mem_support.mp h1 (pow_apply_eq_self_of_apply_eq_self h2 n)
+#align equiv.perm.support_pow_le Equiv.Perm.support_pow_le
+
+@[simp]
+theorem support_inv (σ : Perm α) : support σ⁻¹ = σ.support := by
+  simp_rw [Finset.ext_iff, mem_support, not_iff_not, inv_eq_iff_eq.trans eq_comm, iff_self_iff,
+    imp_true_iff]
+#align equiv.perm.support_inv Equiv.Perm.support_inv
+
+-- Porting note: lower priority to avoid linter complaints about simp-normal form
```

Comment:
Are you sure the linter didn't have legitimate complaints?

Reply:
No, I'm not sure. There were four such cases, let me comment on them specifically.